<?xml version="1.0" encoding="UTF-8"?>
<!--
 Copyright 2025 IHP PDK Authors

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

     https://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
-->
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1070</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <widget class="QTabWidget" name="tabSettings">
      <property name="currentIndex">
       <number>4</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Main</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Top Cell</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbxTopCell"/>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>GDS File</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="txtGdsFile"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnGdsFile">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QWidget" name="wdgSettings" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Substrate</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Substrate File</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="txtSubstrate"/>
            </item>
            <item>
             <widget class="QPushButton" name="btnSubstrate">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="SubstrateView" name="substrateView"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Python</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Run Python Script</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="txtRunPythonScript"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnRunPythonScript">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="PythonEditor" name="editRunPythonScript"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btnGenDefaultPython">
            <property name="text">
             <string>Generate Default</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Ports</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QTableWidget" name="tblPorts">
            <column>
             <property name="text">
              <string>Num</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Voltage</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Z0</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Source Layer</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>From Layer</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>To Layer</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Direction</string>
             </property>
            </column>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QCheckBox" name="cbSubLayerNames">
              <property name="text">
               <string>Use Substrate Layer Names</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="btnRemovePorts">
              <property name="text">
               <string>Remove All</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnReomovePort">
              <property name="text">
               <string>Remove</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnAddPort">
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Simulate</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Output Directory</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="txtRunDir"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnRunDir">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="editSimulationLog"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QComboBox" name="cbxSimTool">
            <item>
             <property name="text">
              <string>OpenEMS</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Palace</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btnStop">
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnRun">
            <property name="text">
             <string>Run</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1070</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionOpen_Python_Model"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuSetup">
    <property name="title">
     <string>Setup</string>
    </property>
    <addaction name="actionPrefernces"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSetup"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockRunControl">
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Run Control</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="lstRunControl">
       <item>
        <property name="text">
         <string>Main</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Substrate</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Python</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Ports</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Simulate</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockLog">
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Log</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTextEdit" name="txtLog"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionPrefernces">
   <property name="text">
    <string>Prefernces...</string>
   </property>
  </action>
  <action name="actionOpen_Python_Model">
   <property name="text">
    <string>Load Python Model...</string>
   </property>
  </action>
  <zorder>dockRunControl</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SubstrateView</class>
   <extends>QGraphicsView</extends>
   <header>substrateview.h</header>
  </customwidget>
  <customwidget>
   <class>PythonEditor</class>
   <extends>QTextEdit</extends>
   <header>pythoneditor.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
